framework/src/wifi_manager/wifi_common.h

```diff
@@ -42,5 +48,131 @@ typedef enum {
+	WIFI_UTILS_CRYPTO_AES,					 /**<  AES encryption                            */
+	WIFI_UTILS_CRYPTO_TKIP,					 /**<  TKIP encryption                           */
+	WIFI_UTILS_CRYPTO_TKIP_AND_AES,			 /**<  TKIP and AES mixed encryption             */
+	WIFI_UTILS_CRYPTO_AES_ENT,				 /**<  Enterprise AES encryption                 */
+	WIFI_UTILS_CRYPTO_TKIP_ENT,				 /**<  Enterprise TKIP encryption                */
+	WIFI_UTILS_CRYPTO_TKIP_AND_AES_ENT,		 /**<  Enterprise TKIP and AES mixed encryption  */
+	WIFI_UTILS_CRYPTO_UNKNOWN,				 /**<  unknown encryption                        */
+} wifi_utils_ap_crypto_type_e;
+
+/**
+ * @brief wifi status (connected, dis_connected, soft_ap)
+ */
+typedef enum {
+	WIFI_UTILS_DISCONNECTED,                 /**<  wifi is disconnected  */
+	WIFI_UTILS_CONNECTED,                    /**<  connected             */
+	WIFI_UTILS_SOFTAP_MODE,                  /**<  soft ap mode          */
+} wifi_utils_status_e;
+
+/**
+ * @brief wifi access point information
+ */
+typedef struct {
+	unsigned int channel;                                    /**<  Radio channel that the AP beacon was received on       */
+	char ssid[WIFI_UTILS_SSID_LEN + 1];                      /**<  Service Set Identification (i.e. Name of Access Point) */
+	unsigned int ssid_length;                                /**<  The length of Service Set Identification               */
+	unsigned char bssid[WIFI_UTILS_MACADDR_STR_LEN + 1];     /**<  MAC address (xx:xx:xx:xx:xx:xx) of Access Point        */
+	unsigned int max_rate;                                   /**<  Maximum data rate in kilobits/s                        */
+	int rssi;                                                /**<  Receive Signal Strength Indication in dBm              */
+	wifi_utils_standard_type_e phy_mode;                     /**< Supported MAC/PHY standard                              */
+	wifi_utils_ap_auth_type_e ap_auth_type;                  /**<  @ref wifi_utils_ap_auth_type                           */
+	wifi_utils_ap_crypto_type_e ap_crypto_type;              /**<  @ref wifi_utils_ap_crypto_type                         */
+} wifi_utils_ap_scan_info_s;
+
+
+/**
+ * @brief wifi ap connect config
+ */
+typedef struct {
+	char ssid[WIFI_UTILS_SSID_LEN + 1];                      /**<  Service Set Identification              */
+	unsigned int ssid_length;                                /**<  Service Set Identification Length       */
+	char passphrase[WIFI_UTILS_PASSPHRASE_LEN + 1];          /**<  ap passphrase(password)                 */
+	unsigned int passphrase_length;                          /**<  ap passphrase length                    */
+	wifi_utils_ap_auth_type_e ap_auth_type;                  /**<  @ref wifi_utils_ap_auth_type            */
+	wifi_utils_ap_crypto_type_e ap_crypto_type;              /**<  @ref wifi_utils_ap_crypto_type          */
+} wifi_utils_ap_config_s;
+
+/**
+ * @brief soft ap mode config
+ */
+typedef struct {
```

Comment:
It is a coding rule violation that type and structure definition are in a line.
Let's define structure first and then define type at next of this.

Reply:
And this is not a rule but usually we use `_s` for structure, `_t` for typedef.

Reply:
Completed. Thanks for the comment.